app/webpacker/components/Tickets/index.jsx

```diff
@@ -8,19 +10,72 @@ import TicketHeader from './TicketHeader';
 import TicketWorkbench from './TicketWorkbench';
 import TicketLogs from './TicketLogs';
 
+function SkateholderSelector({ stakeholderList, setUserSelectedStakeholder }) {
+  const [selectedOption, setSelectedOption] = useState(stakeholderList[0]);
+  return (
+    <>
+      <p>
+        You are part of more than one stakeholder, please select the stakeholder as which you want
+        to visit the ticket page.
+      </p>
+      <Dropdown
+        options={stakeholderList.map((requesterStakeholder) => ({
+          key: requesterStakeholder.id,
+          text: requesterStakeholder.stakeholder.name,
+          value: requesterStakeholder,
+        }))}
+        value={selectedOption}
+        onChange={(__, { value }) => setSelectedOption(value)}
+      />
+      <Button
+        disabled={!selectedOption}
+        onClick={() => setUserSelectedStakeholder(selectedOption)}
+      >
+        Select
+      </Button>
+    </>
+  );
+}
+
 export default function Tickets({ id }) {
   const {
     data: ticketDetails, sync, loading, error,
   } = useLoadedData(actionUrls.tickets.show(id));
 
+  const [userSelectedStakeholder, setUserSelectedStakeholder] = useState();
+  const currentStakeholder = userSelectedStakeholder || ticketDetails?.requester_stakeholders[0];
+  const shouldUserSelectStakeholder = (
+    !userSelectedStakeholder
+    && ticketDetails?.requester_stakeholders?.length > 1);
+
   if (loading) return <Loading />;
   if (error) return <Errored />;
 
   return (
-    <Container fluid>
-      <TicketHeader ticketDetails={ticketDetails} />
-      <TicketWorkbench ticketDetails={ticketDetails} sync={sync} />
-      <TicketLogs logs={ticketDetails.ticket.ticket_logs} />
-    </Container>
+    <>
+      <Container fluid>
+        {userSelectedStakeholder && (
+          <Message>
+            {`You are currently viewing the ticket as stakeholder "${userSelectedStakeholder.stakeholder.name}". If you wish to change, please refresh the page.`}
```

Comment:
Refreshing the page _really_ shouldn't be necessary in React. That's what we have "state" and "useMemo" for. If the dropdown value changes, the rest of the DOM can immediately react to it.

If necessary, move the stakeholder selector higher in the component hierarchy, so that the value is determined first, and then the children can react as needed.

Reply:
Okay I knew react can handle it, but tbh I was just lazy to implement that part 😬 I've done it now. Thank you.